File: app/models/skill.rb
class Skill < ActiveRecord::Base
  validates :name, :category, :description, :link, presence: true
  validates :category, inclusion: {in:['language', 'framework', 'technology', 'miscellaneous']}

  has_many :tags
  has_many :resources, through: :tags

  scope :language, -> { where(category: 'language') }
  scope :framework, -> { where(category: 'framework') }
  scope :technology, -> { where(category: 'technology') }
  scope :miscellaneous, -> { where(category: 'miscellaneous') }

  def self.build_tags(params)
    if params[:tags]
      params[:tags].each do |id|
        binding.pry
        self.last.tags.create(skill_id: id)
      end
    end
  end


end

File: app/assets/javascripts/ng-app/factories/Skill.js
function Skill() {
// debugger
  var Skill = {
      'id': '',
      'name': '',
      'category': '',
      'description': '',
      'link': '',
      'documentation_link': '',
      'version': '',
      'tags': []
    }
  console.log("Skill factory checking in")
  return Skill
}
angular
.module('app')
.factory('Skill', Skill)
